proposals/devcontainer-features.md

```diff
@@ -0,0 +1,142 @@
+| containerEnv | object | A set of name value pairs that sets or overrides environment variables. |
+| privileged | boolean | If preveleged mode is required by the feature. |
+| init | boolean | If its necesarry to run `init`. |
+| capAdd | array | Additional capabilities needed by the feature. |
+| securityOpt | array | Security options needed by the feature. |
+| entrypoint | string | Set if the feature sets requires an entrypoint. |
+
+Options
+
+| Property | Type | Description |
+| :--- | :--- | :--- |
+| id | string | Id of the option |
+| id.type | string | Type of the option |
+| id.enum | string array | possible values |
+| id.default | string | default value for the option |
+| id.description | string | Description for the option |
+
+## devcontainer-collection.json properties.
+
+A feature collection file is a compilation of the `devcontainer-feature.json` files for each individual feature. It inlines all the information in each `devcontainer-feature.json` for each feature.
+
+If the application finds a `devcontainer-collection.json` file in the root of a downloaded tar file then it uses that file as to find the particular feature that will be executed.
+
+In addition to the list of features included, the file includes the following properties.
+
+| Property | Type | Description |
+| :--- | :--- | :--- |
+| name | string | Name of the collection |
+| reference | string | Reference information of the repository and path where the code is stored. |
+| version | string | Version of the code. |
+
+In most cases the `devcontainer-collection.json` file can be generated automatically at the moment of creating a release.
+
+## devcontainer.json properties
+
+Features are referenced in `devcontainer.json`  where the `features` tag consists of an array with the ordered list of features to be included in the container image.
+
+The properties are:
+| Property | Type | Description |
+| :--- | :--- | :--- |
+| id | string | reference to the particular feature to be included. |
+| options | object | Type of the option |
+
+The `id` is the main reference point for how to find and download a particular feature. `id` can be defined in any of the following ways:
+
+| Type | Description |
+| :--- | :--- |
+| feature  | A simple name references a feature included with the application used to execute `devcontainer.json` |
+| organization/repository/{feature or collectionl/feature}/@version | Reference to a particular release in a repository |
+| https://<../URI/..>/devcontainer-features.tgz#{optional feature} | Direct reference to a file download. |
+| ./{local-path}  -or-  ../{local-path} | A path relative to devcontainer.json where a feature or feature collection can be found. |
+
+## Authoring
+
+Features can be authored in a number of languages, the most straight forward being bash scripts. If a feature is authored in a different language information about it should be included in the metadata so that users can make an informed choice about it.
```

Comment:
```suggestion
Features can be authored in a number of languages, the most straightforward being bash scripts. If a feature is authored in a different language, information about it should be included in the metadata so that users can make an informed choice about it.
```